app/dao/fact_billing_dao.py

```diff
@@ -303,29 +305,23 @@ def fetch_billing_data_for_day(process_day, service_id=None):
     current_app.logger.info("Populate ft_billing for {} to {}".format(start_date, end_date))
     transit_data = []
     if not service_id:
-        service_ids = [x.id for x in Service.query.all()]
+        services = Service.query.all()
     else:
-        service_ids = [service_id]
-    for id_of_service in service_ids:
+        services = [Service.query.get(service_id)]
+
+    for service in services:
         for notification_type in (SMS_TYPE, EMAIL_TYPE, LETTER_TYPE):
+            table = get_notification_table_to_use(service, notification_type, process_day, has_delete_task_run=False)
```

Comment:
The `has_delete_task_run` is always `False`. Do we need it?

Reply:
we don't _need_it but i liked this because that way whenever you call this function you have to manually define the value for `has_delete_task_run`, making sure you've thought about it for your function.

Reply:
fair enough